### PR TITLE
Added flight of stairs in js

### PR DESCRIPTION
**Pull request by FireQueen-3010**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

The flight of stairs problem using dp in Javascript

Given two positive integers : n and k. We are currently at stair 0, and want to climb to stair n. We can either climb 1, 2 or 3 stairs at a time, but we can climb 3 stairs only k times. Return the number of ways to climb to the nth stair.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4223 (javascript)

### CODE REVIEW

**Comment by FireQueen-3010** [comment]
Please review @ankitaggarwal23 @HarshCasper

File: JavaScript/dp/flightOfStairs.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+let N=prompt("Enter value of N:");
+let K=prompt("Enter value of K:");
+FlightOfStairs(parseInt(N),parseInt(K));
+
+/*
+Input
+n = 4
+k = 1
+Output
+7
```
Comment (ankitaggarwal23) [inline JavaScript/dp/flightOfStairs.js]: Use proper display message for I/P and O/P and also I checked your code is not displaying output cross check once.

File: JavaScript/dp/flightOfStairs.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Time complexity: O(n*k)
+Space complexity: O(n*k)
+*/
```
Comment (ankitaggarwal23) [inline JavaScript/dp/flightOfStairs.js:55]: Add a new line at the end

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by FireQueen-3010** [comment]
Done with the changes, also regarding the output, since I had to make the input dynamic, in javascript dynamic input works only if parsed using html. For others, who implemented dynamic input using javascript also its the same thing. Please review :) @ankitaggarwal23 @HarshCasper

File: JavaScript/dp/flightOfStairs.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    /*checking k before making calls for 3 steps*/
+    if (k > 0)
+        memo[n][k] += solveUtil(n - 3, k - 1, memo);
+
+    return memo[n][k];
+};
+
+let N=prompt("Enter value of n:");
+let K=prompt("Enter value of k:");
```
Comment (HarshCasper) [inline JavaScript/dp/flightOfStairs.js]: Don't use prompt for taking user input. We are accepting only Node implementations. There are various implementations available in the Javascript directory that can help you to implement this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by FireQueen-3010** [comment]
Done with the requested changes, please review :) @HarshCasper @ankitaggarwal23 
image

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**